tests/test_config.py

```diff
@@ -366,15 +366,29 @@ def testHierarchy(self):
         self.assertEqual(c._D, c2._D)  # Check that the child inherits
         self.assertNotEqual(c2._D, Config._D)
 
-    def testStringYaml(self):
+    def testSerializedString(self):
         """Test that we can create configs from strings"""
 
-        c = Config.fromYaml("""
+        serialized = {
+            "yaml": """
 testing: hello
 formatters:
-  calexp: 3""")
-        self.assertEqual(c["formatters", "calexp"], 3)
-        self.assertEqual(c["testing"], "hello")
+  calexp: 3""",
+            "json": '{"testing": "hello", "formatters": {"calexp": 3}}'
+        }
+
+        for format, string in serialized.items():
+            c = Config.fromString(string, format=format)
+            self.assertEqual(c["formatters", "calexp"], 3)
+            self.assertEqual(c["testing"], "hello")
+
+        # Round trip JSON -> Config -> YAML -> Config -> JSON -> Config
+        c1 = Config.fromString(serialized["json"], format="json")
+        yaml = c1.dump(format="yaml")
+        c2 = Config.fromString(yaml, format="yaml")
+        json = c2.dump(format="json")
+        c3 = Config.fromString(json, format="json")
+        self.assertEqual(c3.toDict(), c1.toDict())
```

Comment:
Would it be useful to add tests for exception when `format` is not recognized or format is different from content (you pass YAML string for `format="json"`)?

Reply:
done. You can always parse JSON with a YAML parser so the default for `fromString()` actually works fine in both cases.